index.bs

```diff
@@ -5465,6 +5465,8 @@ The <dfn id="dfn-string-type" export>string types</dfn> are
 {{DOMString}}, all [=enumeration types=],
 {{ByteString}} and {{USVString}}.
 
+The <dfn export>buffer types</dfn> are {{ArrayBuffer}} and {{SharedArrayBuffer}}.
```

Comment:
So, Web IDL is quite picky about distinguishing types. In particular, a "type" like "string type" or "buffer type" refers to something in the category of types. It does not refer to things in the category of values.

Thus, you really shouldn't say things like "If _bufferSource_ is a buffer type". Because a value can never be a type. A value can be an instance of a type, I guess. (But I don't see much use of that language for the other types defined in this section?)

For example, look at all the reference sites for https://whatpr.org/webidl/1311.html#dfn-numeric-type , and notice how they're specifically talking about types (e.g. return types, overload resolution algorithm internals) and not about values.

Reply:
Is it? The way it defines these types is in terms of objects, no? E.g., "buffer source types" references the same ArrayBuffer the typedef does. I'm not sure we can expect people to understand they're fundamentally different if the fundamental concepts are shared.

https://dontcallmedom.github.io/webdex/b.html#buffer%20source%20types%40%40webidl%25%25dfn shows there is some existing "confusion" around this.

The way I see it the types are comparable to classes and from that perspective and how we deal with those elsewhere in the web platform I think it all works.